packages/web-api-send-notification-runner/src/sender/notification-sender.spec.ts

```diff
@@ -14,30 +27,145 @@ class MockableLogger extends Logger {}
-        id: 'id',
+        scanId: 'id',
         replyUrl: 'replyUrl',
+        runStatus: 'completed',
+        scanStatus: 'pass',
+    };
+
+    const onDemandPageScanResult: OnDemandPageScanResult = {
+        url: 'url',
+        scanResult: null,
+        reports: null,
+        run: {
+            state: 'queued' as OnDemandPageScanRunState,
+            timestamp: 'timestamp',
+        },
+        priority: 1,
+        itemType: ItemType.onDemandPageScanRunResult,
+        id: 'id',
+        partitionKey: 'item-partitionKey',
+        batchRequestId: 'batch-id',
     };
 
     beforeEach(() => {
+        onDemandPageScanRunResultProviderMock = Mock.ofType(OnDemandPageScanRunResultProvider, MockBehavior.Strict);
+        webAPIMock = Mock.ofType(NotificationSenderWebAPIClient);
         loggerMock = Mock.ofType(MockableLogger);
+        promiseUtilsMock = Mock.ofType(PromiseUtils);
         notificationSenderMetadataMock = Mock.ofType(NotificationSenderConfig);
         notificationSenderMetadataMock.setup(s => s.getConfig()).returns(() => notificationSenderMetadata);
 
-        sender = new NotificationSender(notificationSenderMetadataMock.object, loggerMock.object);
+        sender = new NotificationSender(
+            onDemandPageScanRunResultProviderMock.object,
+            webAPIMock.object,
+            notificationSenderMetadataMock.object,
+            loggerMock.object,
+            promiseUtilsMock.object,
+        );
     });
 
-    it('Send Notification', async () => {
-        loggerMock.setup(lm => lm.logInfo(`Id: ${notificationSenderMetadata.id}`)).verifiable(Times.once());
-        loggerMock.setup(lm => lm.logInfo(`Reply URL: ${notificationSenderMetadata.replyUrl}`)).verifiable(Times.once());
+    it('Send Notification Succeeded', async () => {
+        setupReadScanResultCall(onDemandPageScanResult);
+        onDemandPageScanResult.notification = generateNotification(notificationSenderMetadata.replyUrl, 'sent', []);
+
+        setupUpdateScanRunResultCall(onDemandPageScanResult);
+
+        const response = { statusCode: 200 } as ResponseAsJSON;
+
+        webAPIMock
+            .setup(wam => wam.postNotificationUrl(notificationSenderMetadata))
+            .returns(async () => Promise.resolve(response))
+            .verifiable(Times.once());
+
+        await sender.sendNotification();
+    });
+
+    it('Send Notification Failed', async () => {
+        onDemandPageScanResult.notification = undefined;
+        setupReadScanResultCall(onDemandPageScanResult);
+        onDemandPageScanResult.notification = generateNotification(notificationSenderMetadata.replyUrl, 'sendFailed', [
+            { errorType: 'HttpErrorCode', message: 'Bad Request' },
+            { errorType: 'HttpErrorCode', message: 'Bad Request' },
+            { errorType: 'HttpErrorCode', message: 'Bad Request' },
+        ]);
+
+        setupUpdateScanRunResultCall(onDemandPageScanResult);
+
+        promiseUtilsMock
```

Comment:
also verify promiseUtils.delay / system.wait call count is 0, on successful notification call.